src/org/javarosa/model/xform/SMSSerializingVisitor.java

```diff
@@ -131,85 +109,87 @@ public IDataPayload createSerializedPayload(FormInstance model, IDataReference r
      * .DataModelTree)
      */
     public void visit(FormInstance tree) {
-        nodeSet = new String();
 
-        //TreeElement root = tree.getRoot();
-        TreeElement root = tree.resolveReference(rootRef);
+        TreeElement root = tree.getRoot();
 
-        xmlns = root.getAttributeValue("", "xmlns");
-        delimiter = root.getAttributeValue("", "delimiter");
-        if ( delimiter == null ) {
+        delimiter = root.getAttributeValue(NAMESPACE_ODK, "delimiter");
+        if (delimiter == null) {
             // for the spelling-impaired...
-            delimiter = root.getAttributeValue("", "delimeter");
+            delimiter = root.getAttributeValue(NAMESPACE_ODK, "delimeter");
         }
-        prefix = root.getAttributeValue("", "prefix");
+        String prefix = root.getAttributeValue(NAMESPACE_ODK, "prefix");
 
-        xmlns = (xmlns != null)? xmlns : " ";
-        delimiter = (delimiter != null ) ? delimiter : " ";
+        delimiter = (delimiter != null) ? delimiter : " ";
         prefix = (prefix != null) ? prefix : " ";
```

Comment:
Feels to me like the default prefix should be an empty string, not a space. Since we're making broader changes and introducing this in a new class I think we should make improvements of that kind.